src/prop_name.rs

```diff
@@ -0,0 +1,324 @@
+use crate::ffi_util::CStrLike;
+
+use std::ffi::{CStr, CString};
+
+/// A borrowed name of a RocksDB property.
+///
+/// The value is guaranteed to be a nul-terminated UTF-8 string.  This means it
+/// can be converted to [`CStr`] and [`str`] at zero cost.
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[repr(transparent)]
+pub struct PropName(CStr);
+
+impl PropName {
+    /// Creates a new object from a nul-terminated string with no internal nul
+    /// bytes.
+    ///
+    /// Panics if the `value` isn’t terminated by a nul byte or contains
+    /// interior nul bytes.
+    pub(crate) const fn new_unwrap(value: &str) -> &Self {
+        let bytes = if let Some((&0, bytes)) = value.as_bytes().split_last() {
+            bytes
+        } else {
+            panic!("input was not nul-terminated");
+        };
+
+        let mut idx = 0;
+        while idx < bytes.len() {
+            assert!(bytes[idx] != 0, "input contained interior nul byte");
+            idx += 1;
+        }
+
+        // SAFETY: 1. We’ve just verified `value` is a nul-terminated with no
+        // interior nul bytes and since its `str` it’s also valid UTF-8.
+        // 2. Self and CStr have the same representation so casting is sound.
+        unsafe {
+            let value = CStr::from_bytes_with_nul_unchecked(value.as_bytes());
+            &*(value as *const CStr as *const Self)
+        }
+    }
+
+    /// Converts the value into a C string slice.
+    #[inline]
+    pub fn as_c_str(&self) -> &CStr {
+        &self.0
+    }
+
+    /// Converts the value into a string slice.
+    ///
+    /// Nul byte terminating the underlying C string is not included in the
+    /// returned slice.
+    #[inline]
+    pub fn as_str(&self) -> &str {
+        // SAFETY: self.0 is guaranteed to be valid ASCII string.
+        unsafe { std::str::from_utf8_unchecked(self.0.to_bytes()) }
+    }
+}
+
+impl core::ops::Deref for PropName {
+    type Target = CStr;
+
+    #[inline]
+    fn deref(&self) -> &Self::Target {
+        self.as_c_str()
+    }
+}
+
+impl core::convert::AsRef<CStr> for PropName {
+    #[inline]
+    fn as_ref(&self) -> &CStr {
+        self.as_c_str()
+    }
+}
+
+impl core::convert::AsRef<str> for PropName {
+    #[inline]
+    fn as_ref(&self) -> &str {
+        self.as_str()
+    }
+}
+
+impl std::borrow::ToOwned for PropName {
+    type Owned = PropertyName;
+
+    #[inline]
+    fn to_owned(&self) -> Self::Owned {
+        PropertyName(self.0.to_owned())
+    }
+
+    // TODO: Uncomment once compiler version is updated to 1.63.
```

Comment:
Let's bump rust version to uncomment this.

Reply:
Done.